src/main/ruby/truffleruby/core/time.rb

```diff
@@ -228,6 +228,24 @@ def round(places = 0)
     time + (rounded - original)
   end
 
+  def floor(places = 0)
+    original = to_i + subsec.to_r
```

Comment:
Doesn't Time#subsec already return a Rational and Rational#to_r is a no-op?
I see, there is a special case when subsec is zero:
```
> Time.new(0).subsec
=> 0
```